File: Data Structures & Algorithms/jump-game-ii/submission-1.cpp
class Solution {
public:
    int jump(vector<int>& nums) {
        if(nums.size()==1) return 0;
        int mx=0, cs=0, nxtMx=0;
        for(int i=0;i<nums.size();i++) {
            if(i>mx) {
                cs++;
                mx = nxtMx;
            }
            nxtMx = max(nxtMx, i+nums[i]);
            if(nxtMx>=nums.size()-1) return cs+1;
        }
        return -1;
    }
};

File: Data Structures & Algorithms/sort-colors/submission-1.java
class Solution {
    public void sortColors(int[] nums) {
        int l=-1, r=nums.length;

        int i=0;
        while(i<r) {
            if(l<i&&nums[i]==0) {
                l++;
                nums[i] = nums[l];
                nums[l] = 0;
            }
            if(r>i&&nums[i]==2) {
                r--;
                nums[i] = nums[r];
                nums[r] = 2;
            }
            if(nums[i]==1||l>=i) i++;
        }
        return;
    }
}